### PR TITLE
Correct esm/cjs structure

### PR DESCRIPTION
**Pull request by jakearchibald**
Hopefully this fixes https://github.com/jakearchibald/idb-keyval/issues/126.

The intent is to treat `.js` as modules, therefore `.cjs` is used for commonjs.

Hopefully the export map is correct and compatible 😬

### CODE REVIEW

**Comment by philholden** [comment]
Thanks. Copying these changes allowed dev and build to work. Note: equivalent PR also needed for `safari-14-idb-fix` to make things work.

**Comment by jakearchibald** [comment]
Hm, this should include the fix made to `safari-14-idb-fix`

**Comment by philholden** [comment]
My mistake didn't see the 2.0.2 version of safari-14-idb-fix.

File: package.json [hunk 1]
```diff
@@ -2,21 +2,31 @@
...
+      "require": "./dist/cjs/index.cjs",
       "import": "./dist/esm/index.js"
     },
-    "./dist/*": "./dist/*/index.js",
+    "./dist/cjs": "./dist/cjs/index.cjs",
+    "./dist/cjs-compat": "./dist/cjs-compat/index.cjs",
+    "./dist/esm": "./dist/esm/index.js",
+    "./dist/esm-compat": "./dist/esm-compat/index.js",
+    "./dist/iife": "./dist/iife/index-min.js",
+    "./dist/iife-compat": "./dist/iife-compat/index-min.js",
```
Comment (developit) [inline package.json:17]:
```suggestion
    "./dist/iife": "./dist/iife/index.js",
    "./dist/iife-compat": "./dist/iife-compat/index.js",
```

File: package.json [hunk 2]
```diff
@@ -2,21 +2,31 @@
...
     "./package.json": "./package.json"
   },
+  "files": [
+    "dist/**"
```
Comment (developit) [inline package.json:21]:
```suggestion
    "dist"
```

**Comment by jakearchibald** [comment]
Ok, I'm feeling brave

**Comment by jakearchibald** [comment]
This is published as 6.0.1.

**Comment by philholden** [comment]
Thanks, working on this. Project now deploys. Very happy.

**Pull request merged**